docs/tutorial/04-control-snake-movements.md

```diff
@@ -0,0 +1,135 @@
+### 4. Control snake movements
+
+Awesome! Our snake moves fast tile by tile, but we cannot control it yet. Kinda boring, isn't it?
+
+In order to control the snake, we need to handle keyboard inputs when pressing any arrow key and change the snake direction accordingly.
+
+☝️ Remember, the direction is saved in the scene's state as a tuple `{x, y}`.
+
+We need to map all the arrow keys inputs into our direction system:
+
+- when pressing the left arrow key ⬅️, the snake direction needs to be set to `{-1, 0}`.
+- when pressing the right arrow key ➡️, the snake direction needs to be set to `{1, 0}`.
+- when pressing the up arrow key ⬆️, the snake direction needs to be set to `{0, -1}`.
+- when pressing the down arrow key ⬇️, the snake direction needs to be set to `{0, 1}`.
+
+And eventually, we need to ignore any other keypress input (e.g. space bar, return, etc.).
+
+Scenic conveniently provides us the [`handle_input/3`](https://hexdocs.pm/scenic/Scenic.Scene.html#c:handle_input/3) callback to implement it.
+
+> Coach: Explain what a callback is and how it is used for defining behaviours.
+
+`handle_input/3` is invoked when the scene receives an input from a driver. Then, the handler can consume the received input and either propagate it to the scene above in the ViewPort's structure or not.
```

Comment:
```suggestion
`handle_input/3` is invoked when the scene receives input from the underlying hardware. It may consume the received input message or propagate it further up the View Port's supervision structure.
```

I think propagating it up to another "scene" is not entirely right but also not suuuper crucial to explain here. I'd therefore just go for the generic "up the supervision structure" 🤷‍♂ . Aside that, I'm not sure if Scenic's concept of "driver" was yet explained 🤔. Maybe indicating that it's the hardware layer is enough for here? What do you think?

Reply:
> I think propagating it up to another "scene" is not entirely right but also not suuuper crucial to explain here

I agree 👍 
FYI, I copied that from the docs 📖 
> When input arrives at a scene, you can consume it, or pass it along to the scene above you in the ViewPort's supervision structure.

https://hexdocs.pm/scenic/Scenic.Scene.html#c:handle_input/3

Good point! Better not introducing another concept (the `driver`).